### PR TITLE
fix 3213: athena integration column 'year' cannot be resolved

### LINKED ISSUE #3213: COLUMN_NOT_FOUND: line 3:11: Column 'year' cannot be resolved
**Describe the bug**
When deploying OpenCost AWS Cloud Cost integration, after provisioning resources and generating reports, Athena queries fail with error:

```COLUMN_NOT_FOUND: line 3:11: Column 'year' cannot be resolved ```

If I amend the generated WHERE statement to add `date_format` the query executes successfully like so:

```		WHERE (date_format(line_item_usage_start_date, '%Y') = '2025' AND date_format(line_item_usage_start_date, '%m') = '06') AND line_item_usage_start_date >= date '2025-06-16' AND line_item_usage_start_date < date '2025-06-23' AND (line_item_line_item_type = 'Usage' OR line_item_line_item_type = 'DiscountedUsage' OR line_item_line_item_type = 'SavingsPlanCoveredUsage' OR line_item_line_item_type = 'EdpDiscount' OR line_item_line_item_type = 'PrivateRateDiscount')```

Unless I'm missing something in my configuration I believe the generated query needs to be amended to generate the `year` and `date` columns inline.

**To Reproduce**
1. Deploy needed resources per AWS integration guide
2. Connect Athena to CUR output bucket via Glue crawler
3. Deploy OpenCost helm chart
4. Observe Athena query logs for error.

**Expected behavior**
OpenCost acknowledges a cloud integration is deployed and cloud costs are visible in the Cloud Cost dashboard

**Screenshots**

Image

**Which version of OpenCost are you using?**
OpenCost version: 1.115.0 (b0b885e)

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by spa-raj**
## What does this PR change?
*  This PR fixes the athena integration query logic for fetching Cloud Usage Reports (CUR) by introducing a change to date filtering logic.

Updated the `GetPartitionWhere` method to use `date_format` for extracting year and month from the `line_item_usage_start_date` field, ensuring compatibility with the dataset's date format.

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* Users will be able to see their AWS cloud usage reports in the opencost dashboard ui

## Does this PR address any GitHub or Zendesk issues?
* Closes #3213 

## How was this PR tested?
* This PR was tested 
 i. Deploying needed resources per AWS integration guide
 ii. Connecting Athena to CUR output bucket via Glue crawler
 iii. Deploy OpenCost using the Tilt dev tooling.
 iv. Observing Athena query logs for query success.
 v. Verifying the AWS CUR report successfully populating in the opencost dashboard ui.
 
 Dashboard screenshot
image

Tilt costmodel-tests and athena success screenshot
image
image

## Does this PR require changes to documentation?
* No

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
* It should be part of next release but I don't have permissions to label it.

### CODE REVIEW

File: pkg/cloud/aws/athenaintegration.go [hunk 1]
```diff
@@ -329,7 +329,8 @@ func (ai *AthenaIntegration) GetPartitionWhere(start, end time.Time) string {
 	endMonth := time.Date(end.Year(), end.Month(), 1, 0, 0, 0, 0, time.UTC)
 	var disjuncts []string
 	for !month.After(endMonth) {
-		disjuncts = append(disjuncts, fmt.Sprintf("(year = '%d' AND month = '%d')", month.Year(), month.Month()))
+		disjuncts = append(disjuncts, fmt.Sprintf("(date_format(line_item_usage_start_date, '%%Y') = '%d' AND date_format(line_item_usage_start_date, '%%m') = '%02d')",
+			month.Year(), month.Month()))
```
Comment (ameijer) [inline pkg/cloud/aws/athenaintegration.go]: let's freeze a 1.0 version, and make a 2.0 version. can make into an athena config. field added to show what type

**Comment by spa-raj** [comment]
# Major Enhancement: Intelligent CUR 2.0 Partition Detection

 I've significantly enhanced the AWS Athena integration based on extensive research and testing. The latest commits introduce intelligent partition detection that automatically adapts to different CUR configurations.

## Latest Changes

### Dynamic Partition Detection

  - New Method: HasBillingPeriodPartitions() - Dynamically detects if CUR 2.0 tables have billing_period partitions via SHOW PARTITIONS
  - Smart Query Optimization: Automatically uses the most efficient partition format based on actual table structure
  - Enhanced Compatibility: Works with all CUR setups, including edge cases where CUR 2.0 tables lack billing_period partitions

### Improved Partition Strategy

  ```
// Now dynamically chooses the best approach:
  if ai.CURVersion == "1.0" {
      // Uses year/month columns
      "(year = '2024' AND month = '8')"
  } else if hasBillingPeriodPartitions {
      // Uses efficient billing_period partitions  
      "(billing_period = '2024-08')"
  } else {
      // Fallback for tables without billing_period partitions
      "(date_format(line_item_usage_start_date, '%Y') = '2024' AND date_format(line_item_usage_start_date, '%m') = '08')"
  }
```

### Comprehensive Testing

  - **Tested with real data on both CUR 1.0 and CUR 2.0 implementation**
  - Mock Implementation: Added mockAthenaQuerier and mockAthenaIntegration to prevent real AWS API calls during unit tests
  - Complete Test Coverage: All CUR 1.0/2.0 scenarios, cross-year boundaries, fallback cases, and edge conditions
  - Zero Test Failures: All AWS integration tests now pass reliably

#### Technical Deep Dive

  My research revealed that in CUR 2.0:
  - billing_period is a partition metadata column (not a data column) - explains why it doesn't appear in information_schema.columns
  - Some CUR 2.0 setups may not have billing_period partitions depending on configuration
  - Partition detection via SHOW PARTITIONS is the reliable way to determine availability
  - Query performance is significantly better when using partition columns vs. date functions

#### Impact

  - Fixes #3213: Resolves the original "Column 'year' cannot be resolved" error
  - Performance: Uses most efficient querying approach for each CUR configuration
  - Reliability: Robust fallback ensures compatibility across all AWS setups
  - Maintainability: Comprehensive test coverage prevents regressions

#### Ready for Review

The implementation now handles all CUR scenarios intelligently and has been thoroughly tested. This should provide a robust solution for AWS cost data ingestion across different CUR configurations.

File: pkg/cloud/aws/athenaconfiguration.go [hunk 2]
```diff
@@ -190,6 +201,18 @@ func (ac *AthenaConfiguration) UnmarshalJSON(b []byte) error {
...
+	if _, ok := fmap["curVersion"]; ok {
+		curVersion, err := cloud.GetInterfaceValue[string](fmap, "curVersion")
+		if err != nil {
+			return fmt.Errorf("AthenaConfiguration: UnmarshalJSON: %w", err)
+		}
+		ac.CURVersion = curVersion
+	} else {
+		// Default to 2.0 if not specified
+		ac.CURVersion = "2.0"
+	}
```
Comment (ameijer) [inline pkg/cloud/aws/athenaconfiguration.go:214]: @Sean-Holcomb here's the switch statement that we discussed adding last time, think we are good to merge?

**Review by ameijer — APPROVED** [review]

File: pkg/cloud/aws/athenaconfiguration.go (hunk 2 above)
Comment (Sean-Holcomb) [reply pkg/cloud/aws/athenaconfiguration.go:214]: We should default to 1.0, to not break existing users

**Review by Sean-Holcomb — APPROVED** [review]
Thanks for all the tests. Looks good

**Comment by spa-raj** [comment]
@ameijer 2 workflows are pending approval. I guess those are needed for merging.
